### DISCUSSION TITLE
Airflow 2.9.2 pip install with requirements.txt fails with conflicting dependencies

### DISCUSSION THREAD

**Discussion by ptran32** [apache/airflow#40352] [General] [answered]
Hi all,

I created this discussion because my issue did not completely fit any category (either airflow-core or airflow-provider sections)

I'm having an issue with building an Airflow 2.9.2 Docker image locally with extra package using a requirements.txt with pip.

I am using Cloudbuild here in order to extract logs, but I am having the same issue locally, so someone can reproduce with the steps below.

I did not have any issue with all previous Airflow I used (2.8.X, 2.9.1) until 2.9.2.

See successful build [logs](https://gist.github.com/ptran32/32c81430860243d9d8fa903c2e6b0e8b) with 2.9.1
See failing build [logs](https://gist.github.com/ptran32/94285ae8dc8ad36af4f667c6da5c2d45) with 2.9.2

How to reproduce:

Use this [Dockerfile](https://gist.github.com/ptran32/d96aa24d9963cb19738a3baaa049f5eb)
Use this [requirements.txt](https://gist.github.com/ptran32/1160517e8f668a59b40130826da8424a)

Finally, use the command below:
```
docker build -t test-airflow:latest \
--build-arg AIRFLOW_VERSION=2.9.2 \
--build-arg PYTHON_MAJOR_MINOR_VERSION=3.10 \
```

Feel free to reach out if you need additional information's. Thank you for your help.

[note]

The build is really slow compared to my builds with previous versions and I noticed those messages before the error
```
INFO: pip is still looking at multiple versions of apache-airflow-providers-google to determine which version is compatible with other requirements. This could take a while.
```

**Comment by potiuk** [comment, marked as answer]
Follow the description here: https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#upgrading-and-installing-dependencies-including-providers

Installing extra dependencies with constraints of Airlow **Might** or **might not** work - because constarints are there only to provide installation of Airflow in a reproducible way. If you want to add something extra you should have two steps of installation - airflow first (with constraints) and your extra dependencies second (without). Follow the link above.

**Reply by ptran32** [reply]
Thanks for your reply Potiuk,

I'll follow the instructions in the documentation provided
